tests/integration/replica-redirect.tcl

```diff
@@ -93,12 +98,154 @@ start_server {tags {needs:repl external:skip}} {
             } else {
                 fail "Failover from primary to replica did not finish"
             }
+            assert_match *master* [r role]
+            assert_match *slave* [r -1 role]
+
+            assert_error "REDIRECT $replica_host:$replica_port" {$rd read}
+            assert_error "REDIRECT $replica_host:$replica_port" {$rd_blocking read}
+            $rd close
+            $rd_blocking close
+        }
+
+        test {responses in waiting-for-sync state} {
+            r -1 replicaof no one
+            r replicaof $primary_host $primary_port
+            wait_replica_online $primary
+
+            set rd_brpop_before [valkey_deferring_client -1]
+            $rd_brpop_before client capa redirect
+            assert_match "OK" [$rd_brpop_before read]
+            $rd_brpop_before brpop list 0
+
+            wait_for_blocked_clients_count 1 100 10 -1
+            pause_process [srv 0 pid]
+
+            set rd_wait [valkey_deferring_client -1]
+            $rd_wait client capa redirect
+            $rd_wait read ; # Consume the OK reply
+            $rd_wait set foo bar
+            $rd_wait read
+            $rd_wait wait 1 0 ; # Blocks as we can't sync
+
+            # XREAD is a reading command and thus, should
+            # not be redirected if the client is read only.
```

Comment:
This comment does not match the assertion below, which tests for a redirect response